Sources/Profile-View.php

```diff
@@ -1885,33 +1885,38 @@ function TrackIP($memID = 0)
 	}
 	else
 	{
-		$context['ip'] = $user_profile[$memID]['member_ip'];
+		$context['ip'] = ip2range($user_profile[$memID]['member_ip']);
 		$context['base_url'] = $scripturl . '?action=profile;area=tracking;sa=ip;u=' . $memID;
 	}
 
 	// Searching?
 	if (isset($_REQUEST['searchip']))
-		$context['ip'] = trim($_REQUEST['searchip']);
+		$context['ip'] = ip2range(trim($_REQUEST['searchip']));
 
-	if (isValidIP($context['ip']) === false)
+	if (count($context['ip']) != 2)
 		fatal_lang_error('invalid_tracking_ip', false);
 
-	//mysql didn't support like search with varbinary
-	//$ip_var = str_replace('*', '%', $context['ip']);
-	//$ip_string = strpos($ip_var, '%') === false ? '= {inet:ip_address}' : 'LIKE {string:ip_address}';
+	$ip_string = array('{inet:ip_address_low}','{inet:ip_address_high}');
+	$fields = array(
+			'ip_address_low' => $context['ip']['low'],
+			'ip_address_high' => $context['ip']['high'],
+		);
+
 	$ip_var = $context['ip'];
-	$ip_string = '= {inet:ip_address}';
+
+	if ($context['ip']['low'] !=  $context['ip']['high'])
```

Comment:
!==

Reply:
Can you explain?

Reply:
Sorry, in this context, both of your checks can be done using strict type comparision. Good practice.